services/compute/virtual-machines/taxonomy.md

```diff
@@ -0,0 +1,47 @@
+|	CCC-030118	|	Instance Preemptibility	|	Providing the option for using preemptible virtual machine (spot) instances at a lower cost for non-critical or fault-tolerant workloads that may be terminated by the cloud provider after a notice period.	|
+|	CCC-030119	|	Instance Affinity/Anti-affinity	|	Enabling control over the location of virtual machine instances to ensure or prevent co-location on the same physical hardware.	|
+|	CCC-030120	|	Instance Health Checks	|	Providing features for performing health checks on virtual machine instances and automatically replacing or repairing unhealthy instances.	|
+|	CCC-030121	|	Instance Remote Access	|	Offering remote access to virtual machine instances through methods such as SSH or RDP for troubleshooting, debugging, and maintenance purposes.	|
+|	CCC-030122	|	Instance Live Migration	|	The ability to perform live migration of virtual machine instances between physical hosts for maintenance or load balancing purposes without downtime.	|
+|	CCC-030123	|	Instance Remote Configuration	|	Providing tools for remotely configuring virtual machine instances, including deployment automation and configuration management frameworks.	|
+|	CCC-030124	|	Instance Resource Tagging	|	Enabling tagging of virtual machine instances with metadata for organization, management, and cost allocation purposes.	|
+|	CCC-030125	|	Instance Resource Monitoring	|	Providing tools for monitoring resource utilization and performance metrics for virtual machine instances, including CPU usage, memory usage, disk I/O, and network traffic.	|
+|	CCC-030126	|	Custom Images	|	Allows users to create and manage their own customized virtual machine images.	|
+|	CCC-030127	|	Dynamic Performance	|	Providing "burstable" instances for intermittent workloads that accumulate credits during periods of low usage which can be used to burst above baseline performance when needed.	|
+|	CCC-030128	|	Dedicated Instances	|	Providing the option to run instances on physical servers that are dedicated solely to a single customer account, ensuring that the underlying hardware resources are not shared with other customers.	|
+|	CCC-030129	|	Instance Scheduling	|	Offering features for scheduling the startup, shutdown, and maintenance of virtual machine instances based on predefined schedules or conditions.	|
+|	CCC-030130	|	Interoperability with Storage Options	|	Capability to read/write to non-ephemeral external storage including object storage and encrypted block storage.	|
+|	CCC-030131	|	Instance Autoscaling	|	Automatically adjusting the number and instance type of virtual machine instances based on predefined criteria such as CPU utilization or incoming traffic,	|
+|	CCC-030132	|	Instance Grouping	|	Offering logical grouping and management tools for sets of virtual machine instances.	|
```

Comment:
This should live above VMs. (May be they use something like tagging to implement this in VM, which is covered in CCC-030124

Reply:
I was going to just delete this CCC-030132 based on this comment but I double checked, and this seems to be one of those where the implementation differs between providers. The grouping does live above EC2 in AWS, but on GCP the Instance Groups are treated as part of Compute Engine itself: https://cloud.google.com/compute/docs/instance-groups.

That confuses things a bit because the GCP Instance Groups also perform some of the features of an aws auto-scaling group or an azure scale set, but maybe for our purposes maybe we could cover the Instance Groups in the same taxonomy as the other two?

Reply:
Be interested to hear people's thoughts on this. To summarise: if Google treats Instance Groups as 'part of' Compute Engine rather than a separate service, do we need to capture the logical grouping and autoscaling features of Instance Groups as part of the VM taxonomy, or treat them the same as AutoScaling groups/Scale Sets regardless?

Reply:
Can you term everything in terms of a set of compute? If the set contains >1 entry, it's likely referring to GCE whereas if the set has exactly one entry, then it's a plain old VM from any of the hyperscalers.
That should allow the set to own the scaling properties for n>1 but would also allow for scaling sets to exist separately for n=1.
?????

Reply:
Still happy to accept a different solution here but I think this question was largely based on a difference in how these features are _named_ , rather than whether or not the feature is available for VMs on each platform.

For now I've left the 'Instance Grouping' and 'Instance Autoscaling' interfaces in here, which should cover these capabilities while remaining platform agnostic - and still allows Autoscalers to be approached separately with less risk of overlap.